cpp/src/arrow/meson.build

```diff
@@ -238,11 +238,7 @@ if needs_integration or needs_tests
     }
 endif
 
-if needs_json
-    rapidjson_dep = dependency('rapidjson', include_type: 'system')
-else
-    rapidjson_dep = disabler()
-endif
+rapidjson_dep = dependency('rapidjson', include_type: 'system')
```

Comment:
Why do we need to remove the `if needs_json` check?

Reply:
The integration tests requires RapidJSON but does not require the JSON option to be set, so to get that to work without disabling the integration test I think this need to exist without branching

Reply:
OK. Then can we use `if needs_json or needs_integration` instead of removing this condition?

Reply:
No problem - done